### PR TITLE
YQ-2417 show ast in case of compilation error

### PR DESCRIPTION
**Pull request by GrigoriyPA**
### Changelog entry 

Show query ast in case of compilation error

### Changelog category 

* Improvement
* New feature

### Additional information

Added ast for queries with type script and query;
Added ast in cpp sdk for queries

### CODE REVIEW

File: ydb/core/kqp/host/kqp_host.cpp [hunk 1]
```diff
@@ -342,8 +343,18 @@ class TAsyncPrepareYqlResult : public TKqpAsyncResultBase<IKqpHost::TQueryResult
         prepareResult.QueryAst = prepareResult.PreparingQuery->GetPhysicalQuery().GetQueryAst();
     }
 
+    void FillPartialResult(TResult& prepareResult) const override {
+        YQL_ENSURE(QueryCtx->PrepareOnly);
+
+        if (auto exprRoot = GetExprRoot()) {
+            prepareResult.PreparingQuery = std::move(QueryCtx->PreparingQuery);
+            prepareResult.PreparingQuery->MutablePhysicalQuery()->SetQueryAst(KqpExprToPrettyString(*GetExprRoot(), ExprCtx));
```
Comment (UgnineSirdis) [inline ydb/core/kqp/host/kqp_host.cpp:351]: Here is this file in FillResult there is line: 
`prepareResult.QueryAst = prepareResult.PreparingQuery->GetPhysicalQuery().GetQueryAst();`
Why don't we do the same in this function?

File: ydb/core/kqp/host/kqp_host_impl.h [hunk 2]
```diff
@@ -83,6 +85,8 @@ class TKqpAsyncResultBase : public NYql::IKikimrAsyncResult<TResult> {
 protected:
     virtual void FillResult(TResult& result) const = 0;
 
+    virtual void FillPartialResult(TResult&) const {}
```
Comment (UgnineSirdis) [inline ydb/core/kqp/host/kqp_host_impl.h:88]: Can we use FillResult, but check status in it?

File: ydb/core/kqp/compile_service/kqp_compile_actor.cpp [hunk 3]
```diff
@@ -372,6 +372,14 @@ class TKqpCompileActor : public TActorBootstrapped<TKqpCompileActor> {
         PassAway();
     }
 
+    void FillPreparedQuery(std::unique_ptr<NKikimrKqp::TPreparedQuery> preparingQuery, NKikimrKqp::EQueryType queryType) {
```
Comment (UgnineSirdis) [inline ydb/core/kqp/compile_service/kqp_compile_actor.cpp:375]: We fill `KqpCompileResult` in this function. Maybe it's better to name it `FillCompileResult`?

File: ydb/core/kqp/compile_service/kqp_compile_actor.cpp [hunk 4]
```diff
@@ -403,16 +411,10 @@ class TKqpCompileActor : public TActorBootstrapped<TKqpCompileActor> {
...
-                KqpCompileResult->PreparedQuery = preparedQueryHolder;
-                KqpCompileResult->AllowCache = CanCacheQuery(KqpCompileResult->PreparedQuery->GetPhysicalQuery());
-
-                if (AstResult) {
-                    KqpCompileResult->Ast = AstResult->Ast;
-                }
+            FillPreparedQuery(std::move(kqpResult.PreparingQuery), queryType);
+
+            if (AstResult) {
+                KqpCompileResult->Ast = AstResult->Ast;
```
Comment (UgnineSirdis) [inline ydb/core/kqp/compile_service/kqp_compile_actor.cpp:417]: Can we do it in else branch (in case of `status != Ydb::StatusIds::SUCCESS`)?

**Review by UgnineSirdis — CHANGES_REQUESTED** [review]

**Review by Hor911 — APPROVED** [review]

**Review by UgnineSirdis — APPROVED** [review]

**Pull request merged**